rear_camera_py/rear_camera/line_angle_calculator.py

```diff
@@ -226,116 +167,148 @@ def get_transformed_image(self, img: np.ndarray, borderValue=(255, 255, 255)) ->
+                                                          = 80
+                                       射影変換によってできた元画像の枠(直線)と同一とみなす距離
+            distance_threshold (float): 距離閾値(この値より遠い座標は、同一線ではない)
+            canny_th1 (int): Cannyヒステリシス1
+            canny_th2 (int): Cannyヒステリシス2
+            canny_aperture_size (int): Cannyソベルオペレータ アパチャーサイズ
+            do_merge (bool): 増分マージオプション
+
         Returns:
-            img (np.ndarray): マスク処理を行った画像.
+            Union[np.ndarray, None]: 検出した線分、線分を検出できなかった場合は`None`を返す.
         """
-        # 上書き用の画像を作成
-        masked_img = img.copy()
-
-        # グレースケール画像に変換
-        img_gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
-
-        # 画像上の円を検出　戻り値:[[[円の中心点のx座標, 円の中心点のy座標, 円の半径],...]]
-        circles = cv2.HoughCircles(img_gray, cv2.HOUGH_GRADIENT, dp=1.2, minDist=100,
-                                   param1=100, param2=40, minRadius=0, maxRadius=60)
-
-        # 画像上に円が見つからなかった場合
-        if circles is None:
-            return img
-
-        # 中心座標が、画像上側(７割)にある円は考慮しない
-        ignore_border = img.shape[0] * 0.7
-        circles = np.delete(circles[0], np.where(circles[0, :, 1] < ignore_border)[0], axis=0)
-
-        # 考慮する円がない場合
-        if len(circles) == 0:
-            return img
-
-        # 検出した円の半径から上書きする円の半径を決める
-        # NOTE:交点の半径3cm.
-        #      ラインが交わっている部分の長さ1.5cm
-        #      よって上書きする円の半径は、
-        #           検出した交点の半径×1.5
-        #      より大きな半径にする必要がある.
-        circles[:, 2] *= 1.7
-
-        # 整数値かつ16ビットにキャストする ※しないとcv2.circleでerrorが発生
-        circles = np.uint16(np.around(circles))
-
-        # 画像上に円を見つけた場合、円の直径を拡大する
-        for circle in circles:
-            # 円を描画する
-            cv2.circle(
-                masked_img,
-                center=(circle[0], circle[1]),
-                radius=circle[2],
-                color=(255, 255, 255),
-                thickness=-1  # 線の太さ(負の値で内部塗りつぶし)
-            )
-
-        return masked_img
+        # length_threshold_mmをpix値に変換(0.9倍補正)
+        length_threshold_pix = int(self.mm_to_pix(length_threshold_mm) * 0.9)
+
+        # 線分検出モジュールのインスタンス化
+        fld = cv2.ximgproc.createFastLineDetector(
+            length_threshold_pix,
+            distance_threshold,
+            canny_th1,
+            canny_th2,
+            canny_aperture_size,
+            do_merge
+        )
 
-    def pix_to_mm(self, pix: Union[int, float]) -> float:
-        """pixをmmに変換する.
+        # 線分の2点座標を検出　※型:[[[x1,y1,x2,y2]],...]
+        lines = fld.detect(img)
 
-        Args:
-            pix (Union[int, float]): 変換したい値[pix]
+        # 走行体からの距離(detect_dist_from_rbody(mm))までの間(画像に写る)の線分だけ残す
+        img_h, img_w = img.shape[:2]
+        detect_range_from_rbody_mm = 181*2+110  # 検出する範囲 (交点toブロック置き場*2+補正値)
+        _, ignore_border = self.runner_base_coordinate_to_image_base_coordinate_pix(
+            0, detect_range_from_rbody_mm, img_h, img_w)
+        lines = lines[np.where((lines[:, :, 1] > ignore_border) & (lines[:, :, 3] > ignore_border))]
```

Comment:
```suggestion
        lines = lines[np.where((lines[:, :, 1] > ignore_border) & (lines[:, :, 3] > ignore_border))] # 線分のy座標が2点とも境界線より大きい線分を取得 ※型: [[x1,y1,x2,y2],...]
```
個人的に読み辛いって感じたから説明的なコメントあった方がいい気がした。(スルーでもおけ)
型が変わってることは書いといた方がいい